### PR TITLE
create_formatValue_repository

### PR DESCRIPTION
**Pull request by bereket-WMDE**
Bug: [T277057](https://phabricator.wikimedia.org/T277057)

### CODE REVIEW

File: src/data-access/FetchFormatValueRepository.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import FormatValueRepository from '@/data-access/FormatValueRepository';
+import ParseResult from '@/data-access/ParseResult';
+import TechnicalProblem from '@/data-access/errors/TechnicalProblem';
+
+export default class FetchFormatValueRepository implements FormatValueRepository {
+	private readonly forLanguageCode: string;
+	private readonly endpoint: string;
+
+	public constructor( endpoint: string, languageCode: string ) {
```
Comment (micgro42) [inline src/data-access/FetchFormatValueRepository.ts]: This signature expects the endpoint as first and the language code as second argument, but [the actual call of the constructor](https://github.com/wmde/query-builder/pull/236/files#diff-bf287fbde737fe72be832168148747edaf7b96dc7b9da182dcf8d9de0b203be4R18-R21) in `src/ServicesFactory` provides them in the opposite order.

File: tests/unit/data-access/FetchFormatValueRepository.spec.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	it( 'format date from valueData returned from ParseValueRepository', async () => {
+		const repo = new FetchFormatValueRepository(
+			testEndpoint,
+			testLang,
+		);
+		const expectedResult = 'February 1920';
+
+		window.fetch = jest.fn().mockImplementation( () => Promise.resolve( {
+			ok: true,
+			json: async () => ( expectedResult ),
```
Comment (micgro42) [inline tests/unit/data-access/FetchFormatValueRepository.spec.ts]: What is returned by the API is not `February 1920`, but `{"result":"February 1920"}`

File: src/data-access/FetchFormatValueRepository.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+		if ( data.error ) {
+			throw new TechnicalProblem( `${data.error.code}: ${data.error.info}` );
+		}
+
+		return data;
+	}
+
+	public formatValue( dataValue: ParseResult, propertyId: string ): Promise<string> {
+		return this.format( dataValue, propertyId );
```
Comment (micgro42) [inline src/data-access/FetchFormatValueRepository.ts]: I'm confused by this indirection again. Could you explain your thoughts behind it?

File: src/data-access/FetchFormatValueRepository.ts (hunk 1 above)
Comment (bereket-WMDE) [reply src/data-access/FetchFormatValueRepository.ts]: :( thanks for the catch!

File: tests/unit/data-access/FetchFormatValueRepository.spec.ts (hunk 2 above)
Comment (bereket-WMDE) [reply tests/unit/data-access/FetchFormatValueRepository.spec.ts]: fixed

File: src/data-access/FetchFormatValueRepository.ts (hunk 3 above)
Comment (bereket-WMDE) [reply src/data-access/FetchFormatValueRepository.ts]: yea.. like is said in the previous PR, it was a result of copy pasting. which is not great.
I removed this line from both PRs

File: tests/unit/data-access/FetchFormatValueRepository.spec.ts (hunk 2 above)
Comment (micgro42) [reply tests/unit/data-access/FetchFormatValueRepository.spec.ts]: Your _expected result_ is not `{"result":"February 1920"}`, but `"February 1920"`. You need to differentiate between what the API returns and what you want your repository to return.

Currently, the repository is actually directly returning the object it receives from the API.

File: src/data-access/FetchFormatValueRepository.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+			throw new TechnicalProblem( `${response.status}: ${response.statusText}` );
+		}
+
+		const data = await response.json();
+
+		if ( data.error ) {
+			throw new TechnicalProblem( `${data.error.code}: ${data.error.info}` );
+		}
+
+		return data;
```
Comment (micgro42) [inline src/data-access/FetchFormatValueRepository.ts]: Based on the signature, this is supposed to be a string. But currently, it returns the entire API response, which is a object.

**Review by micgro42 — CHANGES_REQUESTED** [review]

File: src/data-access/FetchFormatValueRepository.ts (hunk 4 above)
Comment (bereket-WMDE) [reply src/data-access/FetchFormatValueRepository.ts]: good eye. i guess the test didn't fail here because it was also expecting an object

**Review by micgro42 — APPROVED** [review]

**Pull request merged**